nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Updater/FileWriters/FileWriterWorker.cs

```diff
@@ -292,16 +292,22 @@ internal static ImmutableArray<ProjectDiscoveryResult> GetProjectDiscoveryEvalua
         return projectDiscoveryOrder;
     }
 
-    private async Task<ImmutableArray<string>> TryPerformFileWritesAsync(DirectoryInfo repoContentsPath, FileInfo projectPath, ProjectDiscoveryResult projectDiscovery, ImmutableArray<Dependency> requiredPackageVersions)
+    internal static async Task<ImmutableArray<string>> TryPerformFileWritesAsync(
+        IFileWriter fileWriter,
+        DirectoryInfo repoContentsPath,
+        DirectoryInfo originalDiscoveryDirectory,
+        ProjectDiscoveryResult projectDiscovery,
+        ImmutableArray<Dependency> requiredPackageVersions
+    )
     {
-        var originalFileContents = await GetOriginalFileContentsAsync(repoContentsPath, projectPath.Directory!, [projectDiscovery]);
+        var originalFileContents = await GetOriginalFileContentsAsync(repoContentsPath, originalDiscoveryDirectory, [projectDiscovery]);
```

Comment:
This was the fix: using the initial discovery directory as a base instead of each project file.  The rest was plumbing and renaming.